src/DB/DumpProcessor.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\DB;
+
+use Magento\MagentoCloud\App\GenericException;
+use Magento\MagentoCloud\Config\ConfigException;
+use Magento\MagentoCloud\Config\Database\DbConfig;
+use Magento\MagentoCloud\Cron\JobUnlocker;
+use Magento\MagentoCloud\Cron\Switcher;
+use Magento\MagentoCloud\Filesystem\FileSystemException;
+use Magento\MagentoCloud\Util\BackgroundProcess;
+use Magento\MagentoCloud\Util\MaintenanceModeSwitcher;
+use Magento\MagentoCloud\DB\Data\ConnectionFactory;
+
+/**
+ * Processor for creation database dump
+ */
+class DumpProcessor
+{
+    /**
+     * Database names
+     */
+    const DATABASE_MAIN = 'main';
+    const DATABASE_QUOTE = 'quote';
+    const DATABASE_SALES = 'sales';
+
+    /**
+     * Database name map:
+     * {connection name from environment} => {database name}
+     */
+    const DATABASE_MAP = [
+        ConnectionFactory::CONNECTION_SLAVE => self::DATABASE_MAIN,
+        ConnectionFactory::CONNECTION_QUOTE_SLAVE => self::DATABASE_QUOTE,
+        ConnectionFactory::CONNECTION_SALES_SLAVE => self::DATABASE_SALES,
+    ];
+
+    /**
+     * Connection map:
+     * {connection name from environment} => {magento connection name}
+     */
+    const CONNECTION_MAP = [
```

Comment:
there are too many maps related to split DB. I propose to create one class with one map where all other classes will retrieve ids for env.php or for Split_DB

Reply:
If to create one map, it will be a nested array. Anyway, working with single-level arrays is easier.
To access the values ​​of such a structure, I will most likely need nested constructions "foreach", which is not entirely good and|or additional methods for access to those values.
And there are only two maps

Reply:
I did not mean having one array, - just one place to resolve such mappings. Right now we have 4 mapping arrays related to DB in 3 classes plus these 2 you will add.